src/python/pants/build_graph/address.py

```diff
@@ -313,26 +345,43 @@ def path_safe_spec(self) -> str:
         if parent_prefix == ".":
             target_portion = f"{parent_prefix}{self._target_name}" if self._target_name else ""
         else:
-            target_name = self._target_name or os.path.basename(self.spec_path)
-            target_portion = f"{parent_prefix}{target_name}"
-        return f"{self.spec_path.replace(os.path.sep, '.')}{file_portion}{target_portion}"
+            target_portion = f"{parent_prefix}{self.target_name}"
+        generated_portion = (
+            f"@{self.generated_name.replace(os.path.sep, '.')}" if self.generated_name else ""
+        )
+        return f"{self.spec_path.replace(os.path.sep, '.')}{file_portion}{target_portion}{generated_portion}"
 
-    def maybe_convert_to_build_target(self) -> Address:
-        """If this address is for a file target, convert it back into its BUILD target.
+    def maybe_convert_to_target_generator(self) -> Address:
+        """If this address is generated, convert it to its generator target.
 
         Otherwise, return itself unmodified.
         """
-        if not self.is_file_target:
-            return self
-        return self.__class__(self.spec_path, relative_file_path=None, target_name=self.target_name)
+        if self.is_generated_target or self.is_file_target:
```

Comment:
This `or self.is_file_target` is redundant, as it is already part of the `is_generated_target`.

Reply:
Good catch! Thanks!